indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/autoscaler/LagBasedAutoScaler.java

```diff
@@ -149,7 +149,9 @@ private Runnable computeAndCollectLag()
             long totalLags = lagStats.getTotalLag();
             lagMetricsQueue.offer(totalLags > 0 ? totalLags : 0L);
           }
-          log.debug("Current lags [%s] for dataSource [%s].", new ArrayList<>(lagMetricsQueue), dataSource);
+          if (log.isDebugEnabled()) {
+            log.debug("Current lags [%s] for dataSource [%s].", new ArrayList<>(lagMetricsQueue), dataSource);
+          }
```

Comment:
Sorry I had missed this in the first review.

We can simplify this as follows:
```suggestion
          log.debug("Current lags [%s] for dataSource [%s].", lagMetricsQueue, dataSource);
```

`CircularFifoQueue.toString` and `ArrayList.toString` would return the same string, so no point converting to list.

Reply:
This might also help further with the coverage.